src/main/java/org/zanata/mt/service/TranslationEngine.java

```diff
@@ -0,0 +1,47 @@
+package org.zanata.mt.service;
+
+import java.util.List;
+
+import org.zanata.mt.exception.TranslationEngineException;
+import org.zanata.mt.model.Locale;
+
+/**
+ * Interface for translation engine
+ *
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+public interface TranslationEngine {
+
+    /**
+     * Return raw response from MT provider
+     * @throws TranslationEngineException
+     */
+    String translate(String message, Locale srcLocale,
+            Locale targetLocale) throws TranslationEngineException;
+
+    /**
+     * Return raw response (same index as request) from MT provider
+     * @throws TranslationEngineException
+     */
+    String translate(List<String> messages, Locale srcLocale,
+        Locale targetLocale) throws TranslationEngineException;
+
+    /**
+     * Extract translations from raw response from MT provider
```

Comment:
Not sure I understand the purpose of this method. 

- What is the format of the `response` string?
- What is the format for the returned strings?
- When should it throw an exception?